### DISCUSSION TITLE
Pool cleaner reset

### DISCUSSION THREAD

**Discussion by orrious** [tagyoureit/nodejs-poolController#487] [General]
I'm a long time njs-PC user prior to REM and I ended up building a similar solution as I do not have a commercial controller.  I would however like to standardize on the REM solution and am doing a gap analysis.  

My mx6 pool cleaner gets stuck in different places around the pool. The pool cleaner is on it's own intelivalve and can be turned on and off. I initially created a Node-Red function that, when triggered, will turn the valve off for 30 seconds and back on. Long enough for the cleaner to become unstuck.

All well and good, but I'm lazy and have cameras all over my property. Very high level, I trained openCV to recognize my pool cleaner and track it. If it stays in one position for x seconds, the application will trigger the reset action for me. Now the pool cleaner is automatically unstuck!

I'm happy to share if anyone is interested, but the current cascade classifier will only recognize a MX6 pool cleaner. ( I can show you how to train other data sets for other cleaners.)

Finally to the ask, does nodejs-poolController currently have an api/trigger that will initiate a "cleaner reset" action?

**Comment by rstrouse** [comment]
Tell me more.  I run an MX8 these days it doesn’t get stuck.  However, however njsPC and REM can trigger pretty much anything.

**Reply by orrious** [reply]
I'm calling it poolVision for now.  It's a docker container that is cuda enabled (will use your nvidia gpu).  For now, you provide a pytorch model (this is what you would train for whatever you want to track).  It is currently integrated with Home Assistant via websocket.  I could convert to mqtt very easily.  It listens to the even bus for when the cleaner mode is turned on.  Right now I do all control and automation via HA.  If poolVision detects the cleaner hasn't moved in x number of times, it sends the signal to reset the cleaner.

I would need 3 things:
1.)  njs-PC to have an automation to reset the pool cleaner (if cleaner is on and you get a reset event, then shutoff the valve for x numbers of seconds and turn it back on)
2.) a way to know that the cleaner feature is active (I know this is available via MQTT, but is it via websocket?  What you're preferred integration?
3.) A way to trigger the reset action.

Future features may include monitoring for algae growth / debris to run the cleaner longer or skimmers, etc.

**Reply by rstrouse** [reply]
The MQTT interface is a subset of the web socket interface.  Everything that is available for MQTT is available for websockets
